spritzle/torrent.py

```diff
@@ -0,0 +1,66 @@
+#
+# spritzle/torrent.py
+#
+# Copyright (C) 2016 Andrew Resch <andrewresch@gmail.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3, or (at your option)
+# any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.    See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.    If not, write to:
+#   The Free Software Foundation, Inc.,
+#   51 Franklin Street, Fifth Floor
+#   Boston, MA    02110-1301, USA.
+#
+
+import asyncio
+import logging
+
+log = logging.getLogger('spritzle')
+
+
+class AlertException(Exception):
+    def __init__(self, alert):
+        self.alert = alert
+
+
+class Torrent(object):
+    def __init__(self, core):
+        self.core = core
+        self.loop = asyncio.get_event_loop()
+
+        self.remove_torrent_futures = {}
+        self.delete_torrent_futures = {}
+        self.core.alert.register_handler('torrent_removed_alert', self._on_torrent_removed_alert)
+        self.core.alert.register_handler('torrent_deleted_alert', self._on_torrent_deleted_alert)
+        self.core.alert.register_handler('torrent_delete_failed_alert', self._on_torrent_delete_failed_alert)
+
+    def remove(self, torrent_handle, options=0):
```

Comment:
I'd prefer if we made this a coroutine. Just: return await future_dict[info_hash]

We await this function anyways so I don't see a point in returning a future. If someone doesn't want to wait for this function they can wrap it in an asyncio.ensure_future() call. My guess is that we'll only really call this in one place (resource/torrent.py) anyways.

Reply:
Done.